### PR TITLE
Add static analysis

### PR DESCRIPTION
**Pull request by RobiNino**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
Run a Github action with static analysis and gosec scan on every pr/push (parallel to tests).
Both tools can also be run from the command line.
A few rules are temporarily ignored and will be fixed separately.

### CODE REVIEW

File: artifactory/utils/dotnet/dependencies/assetsjson_test.go [hunk 1]
```diff
@@ -99,6 +99,9 @@ func TestNewAssetsExtractor(t *testing.T) {
 	}
 
 	allDependencies, err := extractor.AllDependencies()
+	if err != nil {
+		t.Error(err)
+	}
```
Comment (yahavi) [inline artifactory/utils/dotnet/dependencies/assetsjson_test.go:104]:
```suggestion
	assert.NoError(t, err)
```

File: artifactory/utils/dotnet/dependencies/packagesconfig_test.go [hunk 2]
```diff
@@ -155,6 +155,9 @@ func TestExtractDependencies(t *testing.T) {
 
 	expectedAllDependencies := []string{"id1", "id2"}
 	allDependencies, err := extractor.AllDependencies()
+	if err != nil {
+		t.Error(err)
+	}
```
Comment (yahavi) [inline artifactory/utils/dotnet/dependencies/packagesconfig_test.go:160]:
```suggestion
	assert.NoError(t, err)
```

File: general/cisetup/pipelinesconfig.go [hunk 3]
```diff
@@ -54,12 +54,10 @@ func (pc *JFrogPipelinesConfigurator) Config() (vcsIntName, rtIntName string, er
...
 func (pc *JFrogPipelinesConfigurator) doAddPipelineSource(psm *pipelines.PipelinesServicesManager, projectIntegrationId int) (err error) {
 	_, err = psm.AddPipelineSource(projectIntegrationId, pc.SetupData.GetRepoFullName(), pc.SetupData.GitBranch, PipelinesYamlName)
-	if err != nil {
-		// If source already exists, ignore error.
-		if _, ok := err.(*services.SourceAlreadyExistsError); ok {
-			log.Debug("Pipeline Source for repo '" + pc.SetupData.GetRepoFullName() + "' and branch '" + pc.SetupData.GitBranch + "' already exists and will be used.")
-			err = nil
-		}
+	// If source already exists, ignore error.
+	if _, ok := err.(*services.SourceAlreadyExistsError); ok {
```
Comment (yahavi) [inline general/cisetup/pipelinesconfig.go:58]: Are you sure about that?

File: general/project/projectinit.go [hunk 4]
```diff
@@ -147,8 +146,8 @@ func (pic *ProjectInitCommand) createBuildMessage(technologiesMap map[coreutils.
 			"\n"
 	}
 	if ok := technologiesMap[coreutils.Docker]; ok {
-		baseurl := strings.TrimLeft(pic.serverUrl, "https://")
-		baseurl = strings.TrimLeft(baseurl, "http://")
+		baseurl := strings.TrimSuffix(strings.TrimSpace(pic.serverUrl), "https://")
+		baseurl = strings.TrimSuffix(baseurl, "http://")
```
Comment (yahavi) [inline general/project/projectinit.go:150]: Shouldn't trim prefix?
```suggestion
		baseurl := strings.TrimPrefix(strings.TrimSpace(pic.serverUrl), "https://")
		baseurl = strings.TrimPrefix(baseurl, "http://")
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Incredible @RobiNino

File: general/project/projectinit.go (hunk 4 above)
Comment (RobiNino) [reply general/project/projectinit.go:150]: Good catch! thanks

**Pull request merged**